internal/databases/mysql/binlog_fetch_handler.go

```diff
@@ -42,7 +43,7 @@ func (handlers BinlogFetchHandlers) DownloadLogTo(logFolder storage.Folder, logN
 		tracelog.ErrorLogger.Print(err)
 		return err
 	}
-	return nil
+	return handlers.applier(dstLogFilePath)
```

Comment:
We do not need to save binlog to local file at all.
Instead we can stream binlog content directly to apply command via pipe.

For instance we could use following WALG_LOG_APPLY_COMMAND:

mysqlbinog - | mysql 

Command should be executed as shell command ($SHELL env variable) here.